packages/core/src/node/messaging/connection.ts

```diff
@@ -44,6 +44,36 @@ export function openSocket(options: IServerOptions, onOpen: OnOpen): void {
         noServer: true,
         perMessageDeflate: false
     });
+    interface ExtWebSocket extends ws {
+        isAlive: boolean;
+    }
+
+    wss.on('connection', (ws: ws) => {
```

Comment:
To avoid warnings due to the shadowed names of `ws`, you could change the argument to `websocket` or whatever that does not clash with `ws`. Also, since you cast the argument, you can remove the `ws` type information too. Something like this:
```typescript
wss.on('connection', websocket => {
  const extWs = websocket as ExtWebSocket;
  extWs.isAlive = true;
});
```

Same [below](https://github.com/theia-ide/theia/pull/1454/files#diff-906d0f57b69cd6a3345df7a0bda73360R59).